RELEASING.md

```diff
@@ -0,0 +1,41 @@
+## Releasing Shopify CLI
+
+1. Check the Semantic Versioning page for info on how to version the new release: http://semver.org
+2. Create a branch named `release_X_Y_Z` (replacing `X_Y_Z` with the intended release version)
+  ```
+  $ git checkout -b release_X_Y_Z
+  ```
+3. Update the version of Shopify CLI in `lib/shopify-cli/version.rb`
+4. Add an entry for the new release to `CHANGELOG.md`
```

Comment:
How granular do we want to keep changelog entries? Do we want to update something there whenever we add / fix a feature, or only when a release is made?

Reply:
Only for releases